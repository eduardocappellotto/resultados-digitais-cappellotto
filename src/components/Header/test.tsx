import { screen } from "@testing-library/react";

import { Header } from ".";
import { renderWithTheme } from "@/utils/tests/helpers";

describe("<Header />", () => {
  it("should render the header with logo and link", () => {
    const { container } = renderWithTheme(<Header />);

    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(
      screen.getByLabelText(/logotipo da resultados digitais/i)
    ).toBeInTheDocument();

    expect(screen.getByRole("link")).toBeInTheDocument();
    expect(screen.getByText(/gerador de cart√£o de visita/i));

    expect(container.firstChild).toMatchSnapshot();
  });
});
