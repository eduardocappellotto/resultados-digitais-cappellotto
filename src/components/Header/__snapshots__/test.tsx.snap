// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the header with logo and link 1`] = `
.c2 {
  max-width: 128rem;
  width: 100%;
  margin: 0 auto;
  padding: 0 2rem;
}

.c3 {
  width: 100px;
  height: 24px;
}

.c0 {
  background-color: #FFFFFF;
  padding: 2.4rem 0;
}

.c0 .c1 {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c0 a {
  font-size: 1.4rem;
  font-weight: 700;
  color: #212429;
  line-height: 1.1;
}

@media (min-width: 768px) {
  .c3 {
    width: 147px;
    height: 36px;
  }
}

@media (min-width: 450px) {
  .c0 a {
    font-size: 1.8rem;
  }
}

@media (min-width: 768px) {
  .c0 a {
    font-size: 2rem;
  }
}

<header
  class="c0"
>
  <div
    class="c1 c2"
  >
    <div
      class="c3"
    >
      <svg
        aria-label="Logotipo da Resultados Digitais"
        fill="none"
        role="img"
        viewBox="0 0 147 36"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M48 6.021h5.054c2.237 0 3.855 1.652 3.855 3.446 0 1.06-.52 1.814-1.019 2.283-.28.286-.619.51-.998.673l1.997 3.446h-2.557l-1.758-2.957h-2.177v2.957H48V6.02Zm4.974 4.608c.959 0 1.438-.49 1.438-1.142 0-.652-.48-1.142-1.438-1.142h-2.577v2.304h2.577v-.02ZM58.507 6.021h7.531v2.304h-5.134v1.386h4.175v2.304h-4.175v1.57h5.134v2.304h-7.53V6.021ZM69.634 12.831c.08.224.2.449.38.632.299.306.838.612 1.777.612 1.359 0 1.838-.49 1.838-.979 0-.55-.7-.795-1.598-1.02l-1.478-.366c-1.558-.347-3.097-1.122-3.097-2.977.02-1.468 1.519-3.038 4.275-3.038 1.878 0 2.957.775 3.576 1.529.36.449.6.938.759 1.509h-2.497c-.06-.163-.18-.306-.32-.449-.26-.244-.699-.469-1.518-.469-1.278 0-1.678.408-1.678.816 0 .795 1.538 1.02 3.097 1.366 1.518.387 3.076 1.142 3.076 2.997 0 1.652-1.518 3.201-4.415 3.201-2.137 0-3.316-.836-3.955-1.672-.38-.489-.64-1.06-.78-1.692h2.558ZM77.744 6.021h2.397v5.668c0 1.305.879 2.223 2.157 2.223 1.279 0 2.158-.898 2.158-2.223V6.021h2.397v5.668c0 2.691-1.758 4.506-4.575 4.506-2.796 0-4.574-1.815-4.574-4.506V6.021h.04ZM88.87 6.021h2.397v7.544h4.975v2.304H88.87V6.02ZM95.283 6.021h8.969v2.304h-3.276v7.544h-2.397V8.325h-3.276V6.021h-.02ZM106.669 6.021h4.734l2.797 9.848h-2.397l-.56-1.978h-4.414l-.56 1.978h-2.397l2.797-9.848Zm3.935 5.587-.879-3.283h-1.358l-.899 3.283h3.136ZM115.318 6.021h3.936c3.336 0 5.453 2.182 5.453 4.934 0 2.773-2.117 4.934-5.433 4.934h-3.956V6.021Zm4.016 7.564c1.777 0 2.796-1.06 2.796-2.63 0-1.57-1.019-2.63-2.796-2.63h-1.599v5.26h1.599ZM131.359 5.695c3.196 0 5.533 2.385 5.533 5.26s-2.317 5.26-5.533 5.26c-3.196 0-5.533-2.385-5.533-5.26s2.317-5.26 5.533-5.26Zm0 8.217c1.678 0 2.956-1.305 2.956-2.957 0-1.651-1.278-2.956-2.956-2.956s-2.956 1.305-2.956 2.956a2.929 2.929 0 0 0 2.956 2.957ZM140.408 12.831c.08.224.2.449.38.632.299.306.838.612 1.777.612 1.359 0 1.838-.49 1.838-.979 0-.55-.699-.795-1.598-1.02l-1.478-.366c-1.558-.347-3.096-1.122-3.096-2.977 0-1.468 1.498-3.038 4.254-3.038 1.878 0 2.957.775 3.576 1.529.36.449.599.938.759 1.509h-2.477c-.06-.163-.18-.306-.319-.449-.26-.244-.7-.469-1.519-.469-1.278 0-1.678.408-1.678.816 0 .795 1.539 1.02 3.097 1.366 1.518.387 3.076 1.142 3.076 2.997 0 1.652-1.518 3.201-4.415 3.201-2.137 0-3.316-.836-3.955-1.672a4.23 4.23 0 0 1-.779-1.692h2.557ZM48 20.314h3.935c3.336 0 5.454 2.181 5.454 4.934 0 2.773-2.118 4.934-5.434 4.934H48v-9.868Zm4.015 7.544c1.778 0 2.797-1.06 2.797-2.63 0-1.57-1.019-2.63-2.797-2.63h-1.598v5.26h1.598ZM58.987 20.314h2.397v9.848h-2.397v-9.848ZM68.495 19.988c2.138 0 3.456.937 4.235 1.855.44.55.78 1.142 1.019 1.835h-2.477c-.14-.265-.34-.49-.58-.693-.439-.347-1.098-.714-2.157-.714-1.678 0-2.956 1.305-2.956 2.957 0 1.651 1.278 2.956 2.956 2.956 1.039 0 1.718-.408 2.157-.816.24-.244.44-.51.58-.815h-1.359v-1.815h5.134v1.815H73.77a5.976 5.976 0 0 1-1.019 1.978c-.779.999-2.117 1.977-4.235 1.977-3.176 0-5.493-2.385-5.493-5.26-.04-2.895 2.297-5.26 5.473-5.26ZM76.126 20.314h2.397v9.848h-2.397v-9.848ZM79.742 20.314h8.969v2.304h-3.276v7.544h-2.397v-7.544h-3.276v-2.304h-.02ZM91.128 20.314h4.734l2.797 9.848h-2.397l-.56-1.978h-4.414l-.56 1.978h-2.397l2.797-9.848Zm3.915 5.586-.879-3.282h-1.358l-.88 3.282h3.117ZM99.777 20.314h2.397v9.848h-2.397v-9.848ZM106.189 27.124c.08.224.2.448.38.632.3.306.839.611 1.778.611 1.358 0 1.838-.489 1.838-.978 0-.55-.7-.795-1.598-1.02l-1.479-.367c-1.558-.346-3.096-1.121-3.096-2.976 0-1.468 1.498-3.038 4.255-3.038 1.878 0 2.956.774 3.576 1.529.359.448.599.938.759 1.509h-2.477c-.06-.164-.18-.306-.32-.449-.26-.245-.699-.469-1.518-.469-1.279 0-1.678.408-1.678.816 0 .795 1.538 1.019 3.096 1.366 1.518.387 3.076 1.141 3.076 2.997 0 1.651-1.518 3.2-4.414 3.2-2.138 0-3.316-.835-3.955-1.671-.38-.49-.64-1.06-.779-1.692h2.556Z"
          fill="#000"
        />
        <g
          clip-path="url(#a)"
        >
          <path
            d="M30.942 13.507 21.33 29.24 24.597 36l9.612-15.732-3.267-6.76ZM9.612 0 0 15.732l3.267 6.76 9.612-15.73L9.612 0ZM23.552 13.507l4.123-6.746-2.642-5.45L24.408 0l-8.262 13.507L6.534 29.24 9.801 36l9.612-15.732-1.074-2.239-2.193-4.522h7.405Z"
            fill="#806EFF"
          />
          <path
            d="m41.6 20.268-3.267-6.76h-7.39l3.266 6.76L24.597 36h7.39L41.6 20.268ZM20.27 6.761 17.003 0h-7.39l3.266 6.761-9.612 15.732h7.39L20.27 6.76Z"
            fill="#000"
          />
          <path
            d="m26.819 20.268 8.247-13.507L31.799 0h-7.39l3.266 6.761-4.123 6.746h-7.406l3.267 6.761L9.801 36h7.405l8.248-13.507 1.365-2.225Z"
            fill="#000"
          />
        </g>
        <defs>
          <clippath
            id="a"
          >
            <path
              d="M0 0h41.6v36H0z"
              fill="#fff"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <nav>
      <a
        href="/"
      >
        Gerador de Cart√£o de Visita
      </a>
    </nav>
  </div>
</header>
`;
